soc/nxp/s32/s32k5/watchdog_init.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/arm/arch.h>
+#include <zephyr/sys/util.h>
+
+/* MCME Register addresses */
```

Comment:
there's `zephyr/soc/nxp/s32/common/mc_me.c`, which could be used (and extended if needed) to perform these operations